### PR TITLE
fix(config-validator): unset `GlobalConfig` class issue

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes

If we are validating global config get the values for `globalOnly` options from the config object rather than `GlobalConfig.get` method.

## Context
When validating `fileConfig` using `conifg-validator`, some options like `hostRules.headers` which depend on `allowedHeaders` could not be validated as `allowedHeaders` was fetched using `GlobalConfig.get` method which has not been set yet

Ref: https://github.com/renovatebot/renovate/pull/27028#discussion_r1486280926

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
File Config:
```json
"hostRules":[{
  headers:{
    "SOME_VAR": "SOME_VALUE"
  },
  "matchHost": "somehost"
}],
"allowedHeaders": ["SOME*"]
}
```

When running config-validator against the above file config:
Before we used to get this an error:
```log
ERROR: Found errors in configuration
       "file": "config.js",
       "errors": [
         {
           "topic": "Configuration Error",
           "message": "hostRules header `SOME_VAR` is not allowed by this bot's `allowedHeaders`."
         }
       ]
 ELIFECYCLE  Command failed with exit code 1.
```

After:
```log
 INFO: Validating renovate.json
 INFO: Validating config.js
 INFO: Config validated successfully
 ```

File: lib/modules/platform/gitea/index.ts [hunk 1]
```diff
@@ -755,6 +756,7 @@ const platform: Platform = {
             body,
             title,
             state: shouldReOpen ? 'open' : (activeIssue.state as IssueState),
+            update_at: DateTime.now().toISO(),
```
Comment (RahulGautamSingh) [inline lib/modules/platform/gitea/index.ts:759]:
```suggestion
```

File: lib/config/validation.spec.ts [hunk 2]
```diff
@@ -1075,5 +1075,75 @@ describe('config/validation', () => {
...
+      };
+      const { warnings, errors } = await configValidation.validateConfig(
+        true,
+        config,
+      );
+      expect(warnings).toHaveLength(0);
+      expect(errors).toHaveLength(0);
+    });
+
+    it('errors if env object is defiend but allowedHeaders is not', async () => {
```
Comment (RahulGautamSingh) [inline lib/config/validation.spec.ts:1128]:
```suggestion
    it('errors if hostRules.headers is defiend but allowedHeaders is not', async () => {
```

**Comment by rarkins** [comment]
Is there any way a user can validate a renovate.json which includes `hostRules.headers`? Or will it look for GlobalConfig every time, get an empty response, and then reject it?

**Comment by RahulGautamSingh** [comment]
I don't think there's any way,when using `config-validator`. They will get an error.

**Comment by rarkins** [comment]
@viceice @secustor wdyt, should we create a feature request where the config validator can take both the global config file as well as the repo config file and do validation?

**Review by rarkins — APPROVED** [review]

**Review by viceice — COMMENTED** [review]
sounds good. we should refactor the validator to use commander. this makes it more easy to add new arguments.

**Review by viceice — APPROVED** [review]

**Pull request merged**